packages/story-editor/src/app/media/uploadQueue/index.ts

```diff
@@ -369,49 +399,68 @@ function useMediaUploadQueue() {
   );
 
   const uploadVideo = useCallback(
-    async (item) => {
+    async (item: QueueItem) => {
       const { id, file, resource, additionalData = {} } = item;
       let { posterFile } = item;
 
       // The newly uploaded file won't have a poster yet.
       // However, we'll likely still have one on file.
       // Add it back so we're never without one.
       // The final poster will be uploaded later by uploadVideoPoster().
-      let newResource = await uploadFile(file, additionalData);
+      let newResource:
+        | ImageResource
+        | VideoResource
+        | GifResource
+        | AudioResource = (await uploadFile(file, additionalData)) as
+        | ImageResource
+        | VideoResource
+        | GifResource
+        | AudioResource;
 
       if (!isMounted.current) {
         return;
       }
 
       // If we don't have a poster yet (e.g. after converting a GIF),
       // try to generate one now.
-      if (!resource.poster && !posterFile) {
+      if (
+        resource.type === ResourceType.Video &&
+        !resource.poster &&
+        !posterFile
+      ) {
         try {
           posterFile = await getFirstFrameOfVideo(file);
         } catch {
           // Not interested in errors here.
         }
       }
+      if (newResource.type === ResourceType.Video && posterFile) {
+        try {
+          const { poster, posterId } = await uploadVideoPoster(
+            newResource.id,
+            posterFile
+          );
 
-      try {
-        const posterFileName = getFileBasename(posterFile);
-        const { poster, posterId } = await uploadVideoPoster(
-          newResource.id,
-          posterFileName,
-          posterFile
-        );
+          if (!isMounted.current) {
+            return;
+          }
 
-        if (!isMounted.current) {
-          return;
-        }
+          newResource = {
+            ...newResource,
+            poster: poster || newResource.poster,
```

Comment:
As far as I can tell following the code, `poster` will always have a value here, so the alternative is never used?